spec/components/candidate_interface/rejection_reasons/reasons_for_rejection_component_spec.rb

```diff
@@ -0,0 +1,92 @@
+        quality_of_application_subject_knowledge_what_to_improve: 'Write in the first person',
+        qualifications_y_n: 'Yes',
+        qualifications_which_qualifications: %w[no_english_gcse no_science_gcse no_degree],
+        performance_at_interview_y_n: 'Yes',
+        performance_at_interview_what_to_improve: 'There was no need to do all those pressups',
+        course_full_y_n: 'No',
+        offered_on_another_course_y_n: 'Yes',
+        offered_on_another_course_details: 'We felt you would be better suited to Mathematics',
+        honesty_and_professionalism_y_n: 'No',
+        safeguarding_y_n: 'No',
+        cannot_sponsor_visa_y_n: 'No',
+        other_advice_or_feedback_y_n: 'Yes',
+        other_advice_or_feedback_details: 'That zoom background...',
+        interested_in_future_applications_y_n: future_applications,
+      }
+    end
+
+    before do
+      allow(application_choice).to receive_messages(provider: provider, course: course)
+    end
+
+    it 'renders rejection reason answers under headings' do
+      result = render_inline(described_class.new(application_choice:))
+      html = result.to_html
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('Something you did:')
+      expect(html).to include('Didn’t reply to our interview offer')
+      expect(html).to include('Shouted a lot')
+      expect(html).to include('Speak calmly')
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('Quality of application:')
+      expect(html).to include('Do not refer to yourself in the third person')
+      expect(html).to include('Write in the first person')
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('Qualifications:')
+      expect(html).to include('No English GCSE grade 4 (C) or above, or valid equivalent')
+      expect(html).to include('No Science GCSE grade 4 (C) or above, or valid equivalent (for primary applicants)')
+      expect(html).to include('No degree')
+      expect(html).not_to include("https://www.find-postgraduate-teacher-training.service.gov.uk/course/#{application_choice.provider.code}/#{application_choice.course_option.course.code}#section-entry")
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('Performance at interview:')
+      expect(html).to include('There was no need to do all those pressups')
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('They offered you a place on another course:')
+      expect(html).to include('We felt you would be better suited to Mathematics')
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('Additional feedback')
+      expect(html).to include('That zoom background...')
+
+      expect(result.css('.govuk-body.app-rejection__label').text).to include('Future applications:')
+      expect(html).to include('The University of Metal would be interested in future applications from you.')
```

Comment:
There's a lot of assertions here, do you think it would it be worth splitting it, at least partially into more it blocks?

Reply:
This is an old file copied into a new namespace. The format is legacy and so I'm not going to spend time on updating it.